crates/autopilot/src/run_loop.rs

```diff
@@ -564,6 +566,14 @@ enum SolveError {
     Failure(anyhow::Error),
 }
 
+#[derive(Debug, thiserror::Error)]
+enum SolutionError {
```

Comment:
Seems like this should be part of the domain now, right?

Reply:
Yes, it can be moved into `domain::competition` where the `Solution` is defined.

https://github.com/cowprotocol/services/pull/2736